Demo/ViewController.swift

```diff
@@ -30,23 +30,30 @@ class ViewController: UIViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+        let startDate = NSDate()
         //: ##Adding UIControl Events
         //:
         //: Old style of setting a target and action for the button
         oldTestButton.addTarget(self, action: #selector(ViewController.tappedSelector(_:)),
                                 forControlEvents: .TouchUpInside)
 
         // This is equivalent to oldTestButton's implementation of setting the action to Tapped
-        testButton.addControlEvent(.TouchUpInside) {
-            self.testButton.setTitle("Tapped!", forState: .Normal)
+        testButton.addControlEvent(.TouchUpInside) { [weak self] in
+            self?.testButton.setTitle("Tapped! @ \(Int(NSDate().timeIntervalSinceDate(startDate)))sec", forState: .Normal)
+            self?.testButton.removeControlEvent(.TouchUpInside);
         }
 
         //: This adds a closure to the second button on the screen to change the text to Tapped2! when being tapped
-        testButton2.addControlEvent(.TouchUpInside, closure: {
-            self.testButton2.setTitle("Tapped2!", forState: .Normal)
-        })
+        testButton2.addControlEvent(.TouchUpInside, closure: { [weak self] in
+            self?.testButton2.setTitle("Tapped2!", forState: .Normal)
+            })
+        
+        // This adds a closure to the second button on the screen to change the text to Tapped2! when being tapped
```

Comment:
The replacement is adding [weak self] to prevent a retain cycle as self is captured within the closure. More on this: http://krakendev.io/blog/weak-and-unowned-references-in-swift